components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -372,13 +369,58 @@ public User getOwner(String tenantUniqueID) throws TenantMgtException {
                 String.format(ERROR_CODE_RESOURCE_NOT_FOUND.getMessage(), tenantUniqueID));
     }
 
+    @Override
+    public void updateOwner(Tenant tenant) throws TenantMgtException {
+
+        int tenantId = tenant.getId();
+        String tenantDomain = tenant.getDomain();
+
+        // Validate the email address.
+        if (StringUtils.isNotBlank(tenant.getEmail())) {
+            try {
+                CommonUtil.validateEmail(tenant.getEmail());
+            } catch (Exception e) {
+                throw new TenantManagementClientException(ERROR_CODE_INVALID_EMAIL.getCode(),
+                        ERROR_CODE_INVALID_EMAIL.getMessage(), e);
+            }
+        }
+
+        TenantInfoBean tenantInfoBean;
+        try {
+            // Validate and update the tenant password before updating the tenant owner.
+            tenantInfoBean = initializeTenantInfoBean(tenantId, tenant);
```

Comment:
Both tenantId, tenant parameters are redundant IMO, tenant will be suffieceint.

Reply:
Here I am using an already existing public method, so not able to change the method parameters